src/utils/swapMinAmounts.ts

```diff
@@ -59,42 +77,73 @@ export async function swapMinAmounts(
   // Get an array of all the swap plugin names
   const swapPluginNamesArr = Object.keys(account.swapConfig)
 
-  // Create a variable to store the swap quotes with the plugin name as the key
-  const minAmountObj: { [key: string]: number } = {}
+  const swapQuoteParamArr: SwapQuoteParam[] = []
 
-  // Loop over each plugin to attempt to get a swap quote
   for (const plugin of swapPluginNamesArr) {
-    try {
-      const disabledPluginMap = createDisabledSwapPluginMap(
-        plugin,
-        swapPluginNamesArr
-      )
-      const requestOptions = { disabled: disabledPluginMap }
-      const swapQuoteAtOrAboveMin = async (
-        nativeAmount: number
-      ): Promise<boolean> => {
-        try {
-          const binarySearchSwapReq = asFromSwapRequest({
-            fromWallet: fromCurrencyWallet,
-            toWallet: toCurrencyWallet,
-            amount: nativeAmount
-          })
-          await account.fetchSwapQuote(binarySearchSwapReq, requestOptions)
-          return true
-        } catch (e) {
-          if (e.message !== 'Amount is too low') throw e
-          return false
-        }
+    swapQuoteParamArr.push({
+      fromCurrencyWallet,
+      toCurrencyWallet,
+      plugin
+    })
+  }
+
+  const minAmountPromiseArr = swapQuoteParamArr.map(async swapQuoteParam => {
+    const { fromCurrencyWallet, toCurrencyWallet, plugin } = swapQuoteParam
+    const disabledPluginMap = createDisabledSwapPluginMap(
```

Comment:
No need to recreate for each plugin. Instead, create one global map.

Reply:
https://github.com/EdgeApp/edge-core-js/blob/c9747591ca36ae95e193ff50c157a45d18123c69/src/types/types.js#L882